CMakeLists.txt

```diff
@@ -145,10 +145,13 @@ zephyr_compile_options(
 )
 
 zephyr_compile_options(
-  $<$<COMPILE_LANGUAGE:ASM>:-xassembler-with-cpp>
   $<$<COMPILE_LANGUAGE:ASM>:-D_ASMLANGUAGE>
 )
 
+# @Intent: Set compiler specific flags related to assembly
```

Comment:
"Set compiler specific flags related to assembly"

From this intent I would assume that it modifed how assembly is assembled and linked.

But I see that it, in part, modifies what kind of optimizations are used.

We need to resolve this mismatch.

It is unnatural to group C language optimizations with assembly defines.

Perhaps a toolchain_cc_disable_optimizations_that_obscure_generated_code() would be appropriate.

Reply:
I am serious.

> we are doing toolchain abstraction here, not abstraction based on language and what level of optimization an option might lead to. ANY option to the compiler might have impact on optimization and code size...

I don't understand what you are saying. Perhaps you can understand me;

"do stuff related to how assembly is generated from the compiler" Is not a useful abstraction. But "disable optimizations that severely obfuscate the generated machine code" is.

Reply:
@mped-oticon : Evidently, CONFIG_READABLE_ASM, although implemented in CMakeLists.txt, does not exist in Kconfig sources.

This code should be removed instead of ported, this will simplify review and porting of toolchain-specific code, as there will be less of it. This is safe to do, as an option that cannot be enabled cannot break anything when it's implementation is removed.

Reply:
yes, let us remove this section inside CONFIG_READABLE_ASM, came in from the Kbuild days and was never used.